tools/redcap2linst.php

```diff
@@ -0,0 +1,399 @@
+    $linst = $dictRecord->toLINST();
+    if (!empty($linst)) {
+        $instruments[$dictRecord->getFormName()][] = $linst;
+    }
+}
+
+// output directory
+$outputDir = $options['outputDir'];
+
+// back-end name/title instrument mapping
+$redcapIntrumentsMap = ($options['redcapConnection'])->getInstruments(true);
+
+fwrite(STDOUT, "\n-- Writing LINST/META files.\n\n");
+
+// write instrument
+foreach ($instruments as $instname => $instrument) {
+    writeLINSTFile(
+        $options['outputDir'],
+        $instname,
+        $redcapIntrumentsMap[$instname],
+        $instrument
+    );
+}
+
+fwrite(STDOUT, "\n-- end\n");
+
+
+// ------------ Functions
+
+/**
+ * Get the dictionary from a CSV file.
+ *
+ * @param string $inputFile             a file path
+ * @param array  $importableInstruments a list of importable instruments
+ * @param bool   $trimName              should the instrument name be trimmed?
+ *
+ * @return RedcapDictionaryRecord[] dictionary
+ */
+function getDictionaryCSVStream(
+    string $inputFile,
+    array $importableInstruments,
+    bool $trimName
+): array {
+    if (empty($inputFile)) {
+        fprintf(STDERR, "Required a REDCap dictionary file.\n");
+        exit(1);
+    }
+
+    // If a local input file was specified just open it
+    $fp = fopen($inputFile, "r");
+    if ($fp === false) {
+        fprintf(STDERR, "Could not open file $inputFile\n");
+        exit(1);
+    }
+
+    // first line = headers, skip them
+    fgetcsv($fp);
+    // use know one instead
+    $headers = RedcapDictionaryRecord::getHeaders();
+
+    // read file
+    $dictionary      = [];
+    $badMap          = 0;
+    $mapped          = 0;
+    $lastREDCapError = '';
+    while ($row = fgetcsv($fp)) {
+        $inst = $row[1];
+        // skip non importable instruments
+        if (!in_array($inst, $importableInstruments, true)) {
+            $msg = " -> instrument not importable '$inst', skipped.\n";
+            // to avoid repeating same msg
+            if ($lastREDCapError !== $msg) {
+                $lastREDCapError = $msg;
+                fwrite(STDERR, $msg);
+            }
+            continue;
+        }
+
+        // metadata
+        $metadata = $row;
+
+        // do not trim
+        if (!$trimName) {
+            $dd = new RedcapDictionaryRecord(zip($headers, $metadata));
+            $mapped++;
+        } else {
+            // try to trim form name
+            try {
+                $dd = new RedcapDictionaryRecord(zip($headers, $metadata), true);
+                $mapped++;
+            } catch (\LorisException $le) {
+                // print error but continue with non trimmed
+                fprintf(STDERR, $le->getMessage());
+                $dd = new RedcapDictionaryRecord(zip($headers, $metadata));
+                $badMap++;
+            }
+        }
+
+        // add to dictionary
+        $dictionary[] = $dd;
+    }
+    fclose($fp);
+
+    // bad map
+    if ($trimName) {
```

Comment:
I think this is not the right condition no ?
```suggestion
    if ($badMap !== 0) {
```